From f4a7494bc57c0a79d905f160a37236bc47c33076 Mon Sep 17 00:00:00 2001
Message-Id: <f4a7494bc57c0a79d905f160a37236bc47c33076.1426052288.git.chang-joon.lee@intel.com>
In-Reply-To: <99f2c84a457bac958678aa60ef795f22948aa382.1426052288.git.chang-joon.lee@intel.com>
References: <99f2c84a457bac958678aa60ef795f22948aa382.1426052288.git.chang-joon.lee@intel.com>
From: Surbhi Kakarya <surbhi.kakarya@intel.com>
Date: Thu, 5 Mar 2015 10:23:29 +0530
Subject: [PATCH 4/5] FOR_UPSTREAM [VPG]: drm/i915: Fix for "Intel" Logo Blink
 during boot-up

CHV has two Display PHY 1/2 and existing code has only
PHY1 Power UP code.
This fix adds support to Power Up PHY2, doing this will
fix the blink/flicker issues during Boot-Up.

Change-Id: I8d4e8da319951abdd44d736908fd7d630f8cd850
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-20671
Signed-off-by: Surbhi Kakarya <surbhi.kakarya@intel.com>
Signed-off-by: Ahn,Henry <henry.ahn@intel.com>
---
 drivers/gpu/drm/i915/intel_pm.c |   14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index c2c99f9..beaff27 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -6709,16 +6709,21 @@ void __vlv_set_power_well(struct drm_i915_private *dev_priv,
 	u32 mask;
 	u32 state;
 	u32 ctrl;
-	enum pipe pipe;
+	enum pipe pipe = INVALID_PIPE;
 
-	if (power_well_id == PUNIT_POWER_WELL_DPIO_CMN_BC) {
+	if (power_well_id == PUNIT_POWER_WELL_DPIO_CMN_BC
+		|| power_well_id == PUNIT_POWER_WELL_DPIO_CMN_D) {
 		if (enable) {
+			if (power_well_id == PUNIT_POWER_WELL_DPIO_CMN_BC)
+				pipe = PIPE_B;
+			else if (power_well_id == PUNIT_POWER_WELL_DPIO_CMN_D)
+				pipe = PIPE_C;
 			/*
 			 * Enable the CRI clock source so we can get at the
 			 * display and the reference clock for VGA
 			 * hotplug / manual detection.
 			 */
-			I915_WRITE(DPLL(PIPE_B), I915_READ(DPLL(PIPE_B)) |
+			I915_WRITE(DPLL(pipe), I915_READ(DPLL(pipe)) |
 				   DPLL_REFA_CLK_ENABLE_VLV |
 				   DPLL_INTEGRATED_CRI_CLK_VLV);
 			udelay(1); /* >10ns for cmnreset, >0ns for sidereset */
@@ -6769,7 +6774,8 @@ out:
 	 * both PLLs disabled, or we risk losing DPIO and PLL
 	 * synchronization.
 	 */
-	if (power_well_id == PUNIT_POWER_WELL_DPIO_CMN_BC && enable)
+	if ((power_well_id == PUNIT_POWER_WELL_DPIO_CMN_BC
+		|| power_well_id == PUNIT_POWER_WELL_DPIO_CMN_D) && enable)
 		I915_WRITE(DPIO_CTL, I915_READ(DPIO_CTL) | DPIO_CMNRST);
 }
 
-- 
1.7.9.5

